From 8f489c24e1770c2852efcaff9419d54bfadf0854 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Fri, 6 Feb 2015 14:56:10 +0100
Subject: [PATCH] mono-context.h: fix build on MIPS

The mono-context.h file uses definitions such as mips_fp and mips_sp,
which are defined in <mono/arch/mips/mips-codegen.h>, but this file is
not included, leading to build failures.

This fix is similar to 4437ecea91bab3fe68ba1c30521acf75052662cd (for
ARM).

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 mono/utils/mono-context.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mono/utils/mono-context.h b/mono/utils/mono-context.h
index d96eb75..ad56e2c 100755
--- a/mono/utils/mono-context.h
+++ b/mono/utils/mono-context.h
@@ -464,6 +464,8 @@ mono_ia64_context_get_fp (MonoContext *ctx)
 
 #elif ((defined(__mips__) && !defined(MONO_CROSS_COMPILE)) || (defined(TARGET_MIPS))) && SIZEOF_REGISTER == 4 /* defined(__ia64__) */
 
+#include <mono/arch/mips/mips-codegen.h>
+
 typedef struct {
 	mgreg_t	    sc_pc;
 	mgreg_t		sc_regs [32];
-- 
2.1.0

